torch/_numpy/_funcs_impl.py

```diff
@@ -305,28 +307,37 @@ def linspace(
 
 
 def geomspace(
-    start: ArrayLike,
-    stop: ArrayLike,
+    start: ArrayLikeOrScalar,
+    stop: ArrayLikeOrScalar,
     num=50,
     endpoint=True,
     dtype: Optional[DTypeLike] = None,
     axis=0,
 ):
     if axis != 0 or not endpoint:
         raise NotImplementedError
-    base = torch.pow(stop / start, 1.0 / (num - 1))
-    logbase = torch.log(base)
+    all_scalars = all(_dtypes_impl.is_scalar_or_symbolic(x) for x in [start, stop, num])
+
+    if all_scalars:
+        power = math.pow
+        log = math.log
+    else:
+        power = torch.pow
+        log = torch.log
+
+    base = power(stop / start, 1.0 / (num - 1))
+    logbase = log(base)
```

Comment:
Note that `num` is always scalar, so this can be written in a more numerically stable way as
```suggestion
    logbase = 1/(num-1) * (log(stop) - log(start))
```
And you can even factor out log(stop) and log(start).